### DISCUSSION TITLE
node-sonos-ts library required

### DISCUSSION THREAD

**Discussion by Liver64** [svrooij/sonos2mqtt#195] [Q&A]
Hey Stephan,
is your node-sonos-ts library required to use sonos2mqtt? I'm asking because i develop Sonos for Loxone based on PHP and would like to publish player status/info via MQTT. I already do, but it is a kind of push service and i would prefer to have a kind of event listener.
Thanks in advance, Oliver

https://github.com/Liver64/LoxBerry-Sonos

btw: I adopted your TV monitoring because this is still a really nice feature, but i did it in full automated way without pre-configuration

**Comment by svrooij** [comment]
@Liver64 this app **sonos2mqtt** does not know anything about sonos speakers or how to control them. It uses node-sonos-ts for that. So if you're using sonos2mqtt you're using node-sonos-ts under the hood. It are two different things (app vs library), so if people don't like my implementation, they can use it in their own node/typescript app and still have all the goodies sonos has to offer.

And most of node-sonos-ts is generated by [my custom generator](https://github.com/svrooij/sonos-api-docs/tree/main/generator/sonos-docs) anyway, which is also used to generate the documentation at https://sonos.svrooij.io

Here are your options:

1. Use sonos2mqtt, which will use node-sonos-ts internally
2. build your own app in node that uses node-sonos-ts
3. Use my [generator](https://github.com/svrooij/sonos-api-docs/tree/main/generator/sonos-docs) to generate a php client for sonos
4. Build your own (using the documentation as [sonos.svrooij.io](https://sonos.svrooij.io)


Ps. Yes the TV feature is nice, should be build into sonos itself.
